### PR TITLE
add static_format parameter to avatar_url_as

### PR DESCRIPTION
**Pull request by khazhyk**
static_format will only apply to static (not animated) avatars. Makes
it easier to grab gif-or-'format' of an avatar. Defaults to 'webp'

### CODE REVIEW

File: discord/user.py [hunk 1]
```diff
@@ -110,8 +110,11 @@ def avatar_url_as(self, *, format, size=1024):
         format: Optional[str]
             The format to attempt to convert the avatar to.
             If the format is ``None``, then it is automatically
-            detected into either 'gif' or 'webp' depending on the
+            detected into either 'gif' or static_format depending on the
             avatar being animated or not.
+        static_format: Optional['str']
```
Comment (Rapptz) [inline discord/user.py]: This isn't `Optional` (this is reserved for parameters that take in `None` as a valid option).

**Comment by khazhyk** [comment]
Fixed comment.

Use case is similar to that of avatar_url or avatar_url_as(format=None), just allowing you to specify non animated format, instead of always using webp

File: discord/user.py [hunk 2]
```diff
@@ -128,7 +131,7 @@ def avatar_url_as(self, *, format, size=1024):
             if self.avatar.startswith('a_'):
                 format = 'gif'
             else:
-                format = 'webp'
+                format = static_format
```
Comment (jhgg) [inline discord/user.py]: png,webp,gif,jpeg are the only supported static formats. everything else will throw a 415 status code. you could save the caller a lot of trouble by validating this and perhaps throwing an exception.

Comment (khazhyk) [inline discord/user.py]: Size seems to also only accept powers of two in range [16, 2048] ?

Comment (khazhyk) [inline discord/user.py]: Done

**Comment by Rapptz** [comment]
Rebased.

**Pull request closed without merging**